pkgs/development/python-modules/docarray/default.nix

```
@@ -0,0 +1,107 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  poetry-core,
+  numpy,
+  orjson,
+  pydantic,
+  rich,
+  types-requests,
+  typing-inspect,
+  # Optional dependencies
+  av,
+  elasticsearch,
+  elastic-transport,
+  fastapi,
+  hnswlib,
+  jax,
+  lz4,
+  jaxlib,
+  pandas,
+  pillow,
+  types-pillow,
+  protobuf,
+  pymilvus,
+  pydub,
+  qdrant-client,
+  redis,
+  smart-open,
+  torch,
+  trimesh,
+  weaviate-client,
+}:
+buildPythonPackage rec {
+  pname = "docarray";
+  version = "0.40.0";
+  format = "pyproject";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-w/O4i3thKMEDCN3b0hZQyYReJw2oUM9nGMsdPYZ9WYY=";
+  };
+
+  nativeBuildInputs = [ poetry-core ];
```

```suggestion
  build-system = [ poetry-core ];
```
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters